compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala

```diff
@@ -1966,8 +1967,6 @@ class QuoteContextImpl private (ctx: Context) extends QuoteContext:
       end extension
     end RecursiveTypeMethodsImpl
 
-    type LambdaType = dotc.core.Types.LambdaType
```

Comment:
What do we gain by removing this? And if a union is preferable but needs to be used in multiple context, maybe we could have `type LambdaType = MethodType | PolyType | TypeLambda` ?